run.sh

```diff
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-cd `dirname $0`
+cd "$(dirname "$0")"
```

Comment:
In general, the `$(...)` syntax is superior to backticks for everything but ease of typing, so I try to use them whenever possible.

if the shebang was bash, it'd be fully portable. And we have already made bash a requirement elsewhere IIRC.